File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [100000 for _ in range(len(nums))]
        dp[0] = 0
        for i in range(len(nums)):
            for j in range(i+1, min(len(nums), i+nums[i]+1)):
                dp[j] = min(dp[j], dp[i]+1)
                    
        return dp[-1]

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int sol = 0;
        int n = nums.length;
        if(n==1)
            return 0;
        int l = 0;
        int r = nums[0];
        while(r<n-1){
            int cur_max = 0;
            for(int i=l; i<=r; i++){
                cur_max = Math.max(cur_max, i+nums[i]);
            }
            l = r+1;
            r = cur_max;
            sol++;
        }
        return sol+1;
    }
}